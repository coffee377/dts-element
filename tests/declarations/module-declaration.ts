import * as dts from '../index';

it('should return correctly with name', () => {
  expect(dts.emit(
    dts.create_module_declaration({
      name: 'M',
    }),
  )).toMatchSnapshot();
});

it('should return correctly with name, children', () => {
  expect(dts.emit(
    dts.create_module_declaration({
      name: 'M',
      children: [
        dts.create_variable_declaration({
          name: 'a',
        }),
        dts.create_function_declaration({
          name: 'b',
        }),
      ],
    }),
  )).toMatchSnapshot();
});
