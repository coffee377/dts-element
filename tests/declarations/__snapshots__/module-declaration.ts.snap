// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correctly with name 1`] = `"module \\"M\\" { }"`;

exports[`should return correctly with name, children 1`] = `
"module \\"M\\" {
    var a: any;
    function b(): any;
}"
`;
